vrouter.go

```diff
@@ -498,20 +514,13 @@ func (self *Vrouter) AddVlan(vlanId uint16, vni uint32, vrf string) error {
 		})
 		if err != nil {
 			log.Errorf("Error creating port vlan flow for vlan %d. Err: %v", vlanId, err)
+			self.agent.vtepTableMutex.RUnlock()
 			return err
 		}
-
-		// Set the metadata to indicate packet came in from VTEP port
-		vrf := self.agent.vlanVrf[vlanId]
-		if vrf == nil {
-			log.Errorf("Invalid vlan to Vrf mapping for %v", *vlan)
-			return errors.New("Invalid vlan to Vrf mapping")
-		}
-
-		vrfid := self.agent.vrfNameIdMap[*vrf]
-
-		if *vrfid == 0 {
-			log.Errorf("Invalid vrf name:%v", *vrf)
+		var vrfid *uint16
+		if vrfid = self.agent.getvrfId(vrf); vrfid == nil {
+			log.Errorf("Invalid vrf name:%v", vrf)
+			self.agent.vtepTableMutex.RUnlock()
```

Comment:
It might be simpler to do a `defer self.agent.vtepTableMutex.RUnlock()` at the top..
might keep the code simpler..